prow/gitattributes/pattern.go

```diff
@@ -1,3 +1,21 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Copied from gopkg.in/src-d/go-git.v4/plumbing/format/gitignore
```

Comment:
What do you mean by this? Can we do that? Doesn't this invalidate the copyright above?

Reply:
Yes it probably does... let me explain.
Right now I am trying to do 2 things:

1. fix the import issue, replicated in #10671
1. work on this PR, so I have "imported" this file to make sure all tests are working **in the meantime**

The reason why I have put the copyright here is because verify_boilerplate wants a copyright (by the way, the year 2019 gives an error too with this script). The initial file doesn't include any copyright.

Reply:
Actually, I could also reimplement the logic in a helper class somewhere in the git package?
I mean, it's probably not the only place in prow where we want to match paths against a "gitignore" pattern...

Reply:
test-infra is using a buggy old version of `verify_boilerplate.py`:
https://github.com/kubernetes/test-infra/blob/716fe2701363982d6f67784f8766377668c1efed/hack/verify_boilerplate.py#L201-L202

(it was fixed in other repos with PRs like https://github.com/kubernetes/repo-infra/pull/8 and https://github.com/kubernetes/kubernetes/pull/60866)